tests/system/src/test/java/test/robot/javafx/stage/CheckWindowOrderTest.java

```diff
@@ -0,0 +1,117 @@
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.robot.javafx.stage;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.scene.Scene;
+import javafx.scene.control.Label;
+import javafx.stage.Modality;
+import javafx.stage.Stage;
+import javafx.stage.Window;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+//See JDK8220272
+public class CheckWindowOrderTest {
+    static Scene scene;
+    static Stage stage;
+    static Stage firstWindow;
+    static Stage secondWindow;
+    static Stage lastWindow;
+    static CountDownLatch startupLatch = new CountDownLatch(4);
+
+    @Test(timeout = 15000)
+    public void topWindowShouldBeTheLast() throws Exception {
+        Thread.sleep(400);
+        Assert.assertTrue("Last Window Should be Focused", lastWindow.isFocused());
+    }
+
+    @BeforeClass
+    public static void initFX() throws Exception {
+        new Thread(() -> Application.launch(TestApp.class, (String[]) null)).start();
+        waitForLatch(startupLatch, 10, "FX runtime failed to start.");
+    }
+
+    @AfterClass
+    public static void exit() {
+        Platform.runLater(() -> {
+            lastWindow.hide();
+            secondWindow.hide();
+            firstWindow.hide();
+            stage.hide();
+        });
+        Platform.exit();
+    }
+
+    public static class TestApp extends Application {
+        @Override
+        public void start(Stage primaryStage) {
+            stage = primaryStage;
+
+            scene = new Scene(new Label("Primary Stage"), 640, 480);
+            primaryStage.setScene(scene);
+            primaryStage.setOnShown(e -> Platform.runLater(startupLatch::countDown));
+            primaryStage.show();
+
+            firstWindow = new TestStage(primaryStage, "First Window");
+            firstWindow.show();
+
+            secondWindow = new TestStage(primaryStage, "Second Window");
+            secondWindow.show();
+
+            lastWindow = openLastWindow(secondWindow);
+            lastWindow.show();
+        }
+
+        TestStage openLastWindow(Window owner) {
+            TestStage stage = new TestStage(owner, "Last Window");
+            stage.initModality(Modality.WINDOW_MODAL);
+
+            return stage;
+        }
+    }
+
+    public static void waitForLatch(CountDownLatch latch, int seconds, String msg) throws Exception {
+        org.junit.Assert.assertTrue("Timeout: " + msg, latch.await(seconds, TimeUnit.SECONDS));
```

Comment:
Minor nit: you can just use `Assert` rather than the fully-qualified class, since it is imported.

Reply:
done!